publication-event-handlers/src/test/java/no/unit/nva/publication/events/handlers/dynamodbstream/ImportCandidateDynamoDbStreamToEventBridgeHandlerTest.java

```diff
@@ -0,0 +1,158 @@
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.OperationType;
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.StreamRecord;
+import com.fasterxml.jackson.databind.JavaType;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import no.unit.nva.model.testing.PublicationGenerator;
+import no.unit.nva.publication.events.bodies.ImportCandidateDataEntryUpdate;
+import no.unit.nva.publication.model.business.Entity;
+import no.unit.nva.publication.model.business.Resource;
+import no.unit.nva.publication.model.business.importcandidate.ImportCandidate;
+import no.unit.nva.s3.S3Driver;
+import no.unit.nva.stubs.FakeContext;
+import no.unit.nva.stubs.FakeEventBridgeClient;
+import no.unit.nva.stubs.FakeS3Client;
+import nva.commons.core.attempt.Try;
+import nva.commons.core.paths.UnixPath;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+class ImportCandidateDynamoDbStreamToEventBridgeHandlerTest {
+
+    public static final String EXPECTED_EXCEPTION_MESSAGE = "expected exception message";
+    private FakeS3Client s3Client;
+    private FakeContext context;
+    private ImportCandidateDynamoDbStreamToEventBridgeHandler handler;
+    private FakeS3Client failingS3Client;
+
+    @BeforeEach
+    public void init() {
+        this.s3Client = new FakeS3Client();
+        failingS3Client = createFailingS3Client();
+        this.context = new FakeContext() {
+            @Override
+            public String getInvokedFunctionArn() {
+                return randomString();
+            }
+        };
+        FakeEventBridgeClient eventBridgeClient = new FakeEventBridgeClient();
+        this.handler = new ImportCandidateDynamoDbStreamToEventBridgeHandler(s3Client, eventBridgeClient);
+    }
+
+    @AfterEach
+    void closeS3Client() {
+        failingS3Client.close();
+    }
+
+    public static Stream<Arguments> dynamoDbEventProvider() {
+        var publication = PublicationGenerator.randomPublication();
+        return
+            Stream.of(Arguments.of(randomEventWithSingleDynamoRecord(Resource.fromPublication(publication), null)),
+                      Arguments.of(randomEventWithSingleDynamoRecord(null, Resource.fromPublication(publication))),
+                      Arguments.of(randomEventWithSingleDynamoRecord(Resource.fromPublication(publication),
+                                                                     Resource.fromPublication(publication))));
```

Comment:
This one might be a bit weak as you've put the same image both times, which I'd hope would be quashed by some "effective changes logic"

Reply:
Maybe I'm saying effective changes should be tested…?